src/Compilers/Core/Portable/Collections/ArrayBuilderExtensions.cs

```diff
@@ -137,6 +137,46 @@ public static bool All<T>(this ArrayBuilder<T> builder, Func<T, bool> predicate)
             }
         }
 
+        /// <summary>
+        /// Maps an array builder to immutable array.
+        /// </summary>
+        /// <typeparam name="TItem"></typeparam>
+        /// <typeparam name="TArg"></typeparam>
+        /// <typeparam name="TResult"></typeparam>
+        /// <param name="items">The sequence to map</param>
+        /// <param name="map">The mapping delegate</param>
+        /// <param name="arg">The extra input used by mapping delegate</param>
+        /// <returns>If the items's length is 0, this will return an empty immutable array.</returns>
+        public static ImmutableArray<TResult> SelectAsArrayWithIndex<TItem, TArg, TResult>(this ArrayBuilder<TItem> items, Func<TItem, int, TArg, TResult> map, TArg arg)
```

Comment:
>SelectAsArrayWithIndex [](http://example.com/codeflow?start=46&length=22)

Minor: Consider naming `SelectAsArray` to match the corresponding `ImmutableArrayExtensions` method.

Reply:
Done.